src/hotspot/share/runtime/abstract_vm_version.cpp

```diff
@@ -121,7 +121,8 @@ const char* Abstract_VM_Version::vm_vendor() {
 #ifdef VENDOR
   return VENDOR;
 #else
-  return "Oracle Corporation";
+  // vm_vendor const char is padded to 64 bytes(63chars+\0) so C object is "comparable"
+  return "Oracle Corporation\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000";
```

Comment:
I suggest:
#define VENDOR "This is my vendor string"

#ifdef VENDOR
static const char vm_vendor_string[64] = VENDOR;
#else
static const char vm_vendor_string[64] = "Oracle Corporation";
#endif

const char* vm_vendor() {
  return vm_vendor_string;
}

Reply:
thanks @stooke @gnu-andrew i'll do some compilation tests. It comes down to how the compiler stores the const VENDOR, i'm hoping since vm_vendor_string is static, it allocates it within a constant SECTION with full 64 bytes, and not do some mv instruction from a "This is my vendor string" section that would be variable by length?

Reply:
@gnu-andrew sorry, the #define was just from my example code - I forgot to delete it.  I was running compiler tests on godbolt.org to make sure I was happy with the behavior.

Reply:
My testing with gcc showed the full 64 bytes were zero allocated, but yes, my worry would be that not all compilers do that.

Reply:
@gnu-andrew this behaviour is specified by the c++ standard (which is not the same as saying any particular compiler adheres to the standard, I know)

https://en.cppreference.com/w/cpp/language/aggregate_initialization#Character_arrays